### PR TITLE
Use active or overdue where applicable

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: app/controllers/api/payments/active_subscriptions_controller.rb [hunk 1]
```diff
@@ -1,5 +0,0 @@
-class API::Payments::ActiveSubscriptionsController < API::BaseController
-  def show
-    render json: AssembleActiveSubscription.(current_user)
-  end
```
Comment (ErikSchierboom) [inline app/controllers/api/payments/active_subscriptions_controller.rb:4]: This should always have been a member on the `SubscriptionsController` class

**Review by ErikSchierboom — COMMENTED** [review]
The main result of this is that we now acknowledge overdue subscriptions in the donation form modals

File: app/models/user.rb [hunk 2]
```diff
@@ -222,16 +222,10 @@ def formatted_reputation(*args)
   end
 
   memoize
-  def active_subscription? = subscriptions.active.exists?
+  def current_subscription? = current_subscription.present?
```
Comment (iHiD) [inline app/models/user.rb:225]: I think we still want `active_subscription` for use in `app/commands/user/insiders_status/determine_eligibility_status.rb`

Comment (iHiD) [reply app/models/user.rb:225]: (In fact, I don't understand how this passes the tests with it removed!)

Comment (ErikSchierboom) [reply app/models/user.rb:225]: `active_subscription` in `determine_eligibility_status` does not use the method on user :)

Comment (iHiD) [reply app/models/user.rb:225]: Ah!

**Review by iHiD — APPROVED** [review]
Go for it!

**Pull request merged**